src/firestore.rules.test.js

```diff
@@ -171,76 +171,186 @@ test('allows scholarship delete when user is admin', async () => {
   );
 });
 
-describe('validation rules - create', () => {
-  const collection = aliceApp.firestore().collection('scholarships').doc();
+describe('scholarship alidation rules', () => {
+  const collection = (id) =>
+    aliceApp.firestore().collection('scholarships').doc(id);
 
-  test('fails when no scholarship name', () => {
-    const scholarship = { ...newScholarship };
-    delete scholarship.name;
+  test('fails create when name not a string', () => {
+    return assertFails(collection('abc').set({ ...newScholarship, name: 34 }));
+  });
 
-    return assertFails(collection.set(scholarship));
+  test('fails update when name not a string', async () => {
```

Comment:
Oh, I didn't mean make a corresponding update test for each one. But rather just make sure `validData()` is run for update operations (could've just been a single test).

But if you wanted to do this you could also probably refactor this so that firestore is prepopulated with an existing scholarship and just change the scholarship used based on whether it's existing or not. There are probably other refactoring opportunities for similar scholarships.

But these refactors are not really high priority right now. Better to address them later when we convert everything to TypeScript.

Reply:
lol cool I'll leave as is for now.